src/resources/basis.js

```diff
@@ -1,518 +1,320 @@
-        // available or fails
-        const performHttpDownload = function () {
-            http.get(
-                wasmUrl,
-                { cache: true, responseType: "arraybuffer", retry: false },
-                function (err, result) {
-                    if (result) {
-                        WebAssembly.compile(result)
-                            .then(function (result) {
-                                compiledModule = result;
-                                downloadCompleted();
-                            });
-                    }
-                });
-        };
-
-        // download and compile wasm module
-        if (WebAssembly.compileStreaming) {
-            WebAssembly.compileStreaming(fetch(wasmUrl))
-                .then(function (result) {
-                    compiledModule = result;
-                    downloadCompleted();
-                })
-                .catch(function (reason) {
-                    console.error(reason);
-                    console.warn('compileStreaming() failed for ' + wasmUrl + ', falling back to arraybuffer download...');
-                    // failed to stream download, attempt arraybuffer download
-                    performHttpDownload();
-                });
-        } else {
-            performHttpDownload();
-        }
-
-        // download glue script
-        http.get(
-            glueUrl,
-            { cache: true, responseType: "text", retry: false },
-            function (err, result) {
-                glueCode = result;
-                downloadCompleted();
-            });
-    } else {
-        // download fallback script
-        http.get(
-            fallbackUrl,
-            { cache: true, responseType: "text", retry: false },
-            function (err, result) {
-                if (result) {
-                    basisInitialize(result, null, callback);
-                }
-            });
+// global state
+const queue = new BasisQueue();
+let initializing = false;
+
+/**
+ * @name basisInitialize
+ * @function
+ * @description Initialize the Basis transcode worker.
+ * @param {object} [config] - The Basis configuration.
+ * @param {string} [config.glueUrl] - URL of glue script.
+ * @param {string} [config.wasmUrl] - URL of the wasm module.
+ * @param {string} [config.fallbackUrl] - URL of the fallback script to use when wasm modules aren't supported.
+ * @param {number} [config.numWorkers] - Number of workers to use for transcoding (default is 1). While it is
+ * possible to improve transcode performance using multiple workers, this will likely depend on the runtime
+ * platform. For example, desktop will likely benefit from more workers compared to mobile. Also
+ * keep in mind that it takes time to initialize workers and increasing this value could impact application
+ * startup time. Make sure to test your application performance on all target platforms when changing this parameter.
```

Comment:
Is there a maximum number we allow here? I can see from the code we don't clamp and upper bound. Should we?

Reply:
I'm not aware of any limit imposed by the underlying system and don't particularly see a risk leaving it unbounded, but I am also happy to place an upper limit if you think it prudent. What number would you suggest?

Reply:
I'm not sure how I feel about a limit. But if I did impose one, maybe 16?

Reply:
Added limit of 16.